Keas.Core/Models/KeyPartial.cs

```diff
@@ -0,0 +1,27 @@
+using System.ComponentModel.DataAnnotations;
+
+namespace Keas.Core.Models
+{
+    public class KeyDTO
+    {
+        public int Id { get; set; }
+        public string Code { get; set; }
+        public string Name { get; set; }
+        public int TeamId { get; set; }
+        public string Tags { get; set; }
+    }
+
+    public class KeyPartial
+    {
+        [Required]
+        public int Id { get; set; }
+        
+        public KeyDTO Key { get; set; }
+
+        public int SpacesCount { get; set; }
+
+        public int SerialsInUseCount { get; set; }
+ 
+        public int SerialsTotalCount { get; set; }
+    }
```

Comment:
Is there a specific reason for splitting this info into two classes? If not, I'd recommend merging them into a single flat class.

Reply:
I'm using a DTO for the Key since there are some elements of the Key that we don't need to return back.

Reply:
But why not have your KeyDTO include SpacesCount, SerialsInUseCount and SerialsTotalsCount?

Reply:
@jSylvestre wanted the model to look like:
```
key = new Key
{
Id = r.Id,
Code = r.Code,
Name = r.Name,
Notes = r.Notes,
TeamId = r.TeamId,
Tags = r.Tags
},
id = r.Id,
SpacesCount = r.SpacesCount,
SerialsInUseCount = r.SerialsInUseCount,
SerialsTotalCount = r.SerialsTotalCount
```
So I left them outside of the DTO. Is it better to have them inside @sprucely ?

Reply:
I think it's cleaner, but @jSylvestre may have a reason for keeping them separate.

Reply:
@jSylvestre Thoughts about this?

Reply:
@wongband I don't have an opinion on it. Maybe check with @srkirkland

Reply:
I don't see why you would need the KeyDTO part since Key already covers the same properties, but you do need a new object with Key as a prop in order to add on the additional fields

Reply:
@srkirkland The reason why we added the KeyDTO is because the `Key` Class has attributes that are not needed such as KeyXSpaces

Reply:
Ok then I'm for it -- maybe put the models into a folder "Api" or some such so we know it's only used for that purpose.

Reply:
Okay sounds good. Should we add: 
```
        public int SpacesCount { get; set; }

        public int SerialsInUseCount { get; set; }

        public int SerialsTotalCount { get; set; }
```
in the KeyDTO?

Reply:
From what I can tell, that won't produce the correct output structure, but if it does then I'm for it.  My guess is you are going to need a class KeyWithCounts class which has the KeyDTO as one property.

Reply:
Since the return data looks like:
```
{
  "key": {
    "code": "ER",
    "keyXSpaces": null,
    "serials": null,
    "title": "Test",
    "id": 23,
    "name": "Test",
    "notes": null,
    "tags": "Student",
    "team": null,
    "teamId": 10,
    "active": true
  },
  "id": 23,
  "spacesCount": 0,
  "serialsInUseCount": 2,
  "serialsTotalCount": 2
}
```
I think it looks right to have them outside